source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -153,14 +155,14 @@ void AuthenticatorImpl::startVerify() {
     return;
   }
 
-  // TODO(qiwzhang): Cross-platform-wise the below unix_timestamp code is wrong as the
-  // epoch is not guaranteed to be defined as the unix epoch. We should use
-  // the abseil time functionality instead or use the jwt_verify_lib to check
-  // the validity of a JWT.
   // Check "exp" claim.
-  const uint64_t unix_timestamp =
-      std::chrono::duration_cast<std::chrono::seconds>(timeSource().systemTime().time_since_epoch())
-          .count();
+  // We use the current system time and allow for up to 5 seconds of slack.
+  // Consider the following case:
+  // AWS ALB receives a request and determines that the existing access token is valid 
+  // (for another 1 seconds), forwards the request to Istio Ingressgateway and subsequently 
+  // to some pod with an envoy sidecar. Meanwhile, 0.1 seconds have passed and when envoy checks 
+  // the token it finds that it has expired.
+  const uint64_t unix_timestamp = absl::ToUnixSeconds(absl::Now()) - 5;
```

Comment:
I am fine with change to use absl::ToUnixSeconds(absl::Now()).

But I don't like "- 5" part, it seems like a hack to solve your particular problem

BTW,  even you passed this check, there is another check in verifyJwt which is calling this code here
https://github.com/google/jwt_verify_lib/blob/master/src/verify.cc#L163

Reply:
The specific amount of slack is debatable. However, I don't think there is any doubt that there is a general need for some slack. Whatever part of an authn system (be it AWS ALB or https://istio.io/blog/2019/app-identity-and-access-adapter/) that refreshes access tokens has to check if the current token is still valid. Finally, istio-proxy validates the token again and the time between these two validation events is the time needed as slack.

Reply:
Could we make the amount configurable?

Reply:
Yes, I like this idea.  Let us make it configurable from filter config.